pkg/inject/inject.sh

```diff
@@ -78,6 +78,12 @@ if {{ .ExistsCheck }}; then
   sh_c='sh -c'
   if [ "$user" != 'root' ]; then
     if command_exists sudo; then
+      # check if sudo requires a password
+      if echo $(sudo -n uptime 2>&1) | grep -q "a password is required"; then
```

Comment:
I think here we can simply check the return code:

`if ! sudo -n uptime; then`

Should work the same way

Reply:
I think we need to make sure the command doesn't output anything on stdout or stderr as otherwise the communication could get broken

Reply:
Yes you can redirect both strerr and out to dev/null, but the thing to check is more the return code than the output (that might change in some ways in the future)

Reply:
Also rather than using `uptime`, doing `sudo -nl` (list available commands, see the "EXIT VALUE" section of man sudo) is cleaner (it avoids a log about the use of sudo, as it's clearly a no-op command). Leading to:

```shell
if ! sudo -nl >/dev/null 2>&1; then
```

Reply:
thanks, that's tidy. using that